rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -401,19 +401,18 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
                 Ok(Arc::new(builder.finish()) as ArrayRef)
             }
             Date64(DateUnit::Millisecond) => {
-                use chrono::{NaiveDate, NaiveTime};
-                let zero_time = NaiveTime::from_hms(0, 0, 0);
+                use chrono::NaiveDateTime;
                 let string_array = array.as_any().downcast_ref::<StringArray>().unwrap();
                 let mut builder = PrimitiveBuilder::<Date64Type>::new(string_array.len());
                 for i in 0..string_array.len() {
                     if string_array.is_null(i) {
                         builder.append_null()?;
                     } else {
-                        match NaiveDate::parse_from_str(string_array.value(i), "%Y-%m-%d")
-                        {
-                            Ok(date) => builder.append_value(
-                                date.and_time(zero_time).timestamp_millis() as i64,
-                            )?,
+                        match NaiveDateTime::parse_from_str(
+                            string_array.value(i),
```

Comment:
See latest commits in #8913. There is a default parser that also avoids parsing the format for each value, and also supports the decimal fraction

Reply:
Thanks @Dandandan will update.

Reply:
have updated with your suggested changes. thank you very much - this helps me learn.